database/rrdhost.c

```diff
@@ -104,31 +104,33 @@ inline RRDHOST *rrdhost_find_by_hostname(const char *hostname) {
     return dictionary_get(rrdhost_root_index_hostname, hostname);
 }
 
+static inline void rrdhost_index_del_hostname(RRDHOST *host) {
+    if(unlikely(!host->hostname)) return;
+
+    if(rrdhost_option_check(host, RRDHOST_OPTION_INDEXED_HOSTNAME)) {
+        if(!dictionary_del(rrdhost_root_index_hostname, rrdhost_hostname(host)))
+            error("RRDHOST: %s() failed to delete hostname '%s' from index", __FUNCTION__, rrdhost_hostname(host));
+
+        rrdhost_option_clear(host, RRDHOST_OPTION_INDEXED_HOSTNAME);
+    }
+}
+
 static inline RRDHOST *rrdhost_index_add_hostname(RRDHOST *host) {
     if(!host->hostname) return host;
 
     RRDHOST *ret_hostname = dictionary_set(rrdhost_root_index_hostname, rrdhost_hostname(host), host, sizeof(RRDHOST));
     if(ret_hostname == host)
         rrdhost_option_set(host, RRDHOST_OPTION_INDEXED_HOSTNAME);
     else {
-        rrdhost_option_clear(host, RRDHOST_OPTION_INDEXED_HOSTNAME);
-        error("RRDHOST: %s() host with hostname '%s' is already indexed", __FUNCTION__, rrdhost_hostname(host));
+        //have the same hostname but it's not the same host
+        //keep the new one
+        rrdhost_index_del_hostname(ret_hostname);
```

Comment:
Probably need to check that the host we are trying to remove is ORPHAN / ARCHIVED. This is probably needed to avoid having for example 3 children with the same hostname and two of them trying to connect

Reply:
I will yes. I actually had a check that the new one wasn't archived, but didn't work (unseting that flag happens later during stream).